tests/test_rawpy.py

```diff
@@ -1,48 +1,97 @@
 """Tests for rawpy plugin
 """
 
-import rawpy
 import imageio.v3 as iio
+
+import pytest
+import rawpy
 import numpy as np
 
 
-def test_nef_local(test_images):
+@pytest.mark.parametrize(
+    "im_in",
+    [
+       ("Nikon_uncompressed.nef"),
+       ("Blackmagic.dng"), 
+    ],
+)
+def test_im_remote(test_images, im_in):
     """Test for reading .nef file from .test_images dir.
     """
     # Construct image path
-    im_path = test_images / "infrared.nef"
+    im_path = test_images / im_in
 
     # Test if plugin's content mathces rawpy content
     actual = iio.imread(im_path, index=..., plugin="rawpy")
     expected = rawpy.imread(str(im_path)).postprocess()
     assert np.allclose(actual, expected)
 
 
+@pytest.mark.parametrize(
+    "im_in",
+    [
+       ("camera.png"),
+       ("bricks.jpg"), 
+    ],
+)
+def test_unsupported_image_type(test_images, im_in):
+    """Test for unsupported image file.
+    """
+    # Construct image path
+    im_path = test_images / im_in
+
+    with pytest.raises(Exception):
+        iio.imread(im_path, index=..., plugin="rawpy")
+
+
+def test_write(test_images, tmp_path):
+    """Test for writing .nef file from .test_images dir.
+    """
+    # Construct image path
+    im_path = test_images / "Blackmagic.dng"
+
+    # the base image as numpy array
+    im_actual = iio.imread(im_path, index=..., plugin="rawpy")
+    iio_file = tmp_path / "test.png"
+    with pytest.raises(NotImplementedError):
+        iio.imwrite(iio_file, im_actual, plugin="rawpy")
```

Comment:
Writing is not supported, so we can actually skip this test and (if necessary) exclude the `raise NotImplementError` line from coverage.

Reply:
The test has been removed as per our last discussion. Thanks for the review.